docker-app/qfieldcloud/core/views/files_views.py

```diff
@@ -269,6 +269,11 @@ def post(self, request, projectid, filename, format=None):
 
         is_the_qgis_file = utils.is_the_qgis_file(filename)
 
+        if project.is_shared_datasets_project and is_the_qgis_file:
```

Comment:
```suggestion
        if is_the_qgis_file and project.is_shared_datasets_project:
```

Only because all the `if`s down there first check for `is_the_qgis_file` and then for the rest.